internal/ir/xds.go

```diff
@@ -1784,6 +1815,26 @@ func (o *JSONPatchOperation) IsJSONPathNilOrEmpty() bool {
 	return o.JSONPath == nil || *o.JSONPath == EmptyPath
 }
 
+func (o *JSONPatchOperation) Validate() error {
```

Comment:
can we add a single IR translation test with all invalid cases?

Reply:
I've added a unit-test and made the `validate` method more aligned with RFC 6902

Reply:
I also tried to add CEL validation to the JSONPatchOperation structure in the API folder. 

Turns out that because the `Value` field there is `apiextensionsv1.JSON`, it's not possible to reference the `value` field in CEL tests. Any CEL expression that tries doesn't compile correctly. So no CEL tests on the `JSONPatchOperation` are possible if they reference the `value` attribute.